### DISCUSSION TITLE
Navigation (SSR) - Is there a way to avoid getInitialProps being called repeatedly

### DISCUSSION THREAD

**Discussion by icew1nd** [vercel/next.js#17459] [Help] [answered]
Hi everyone :wave: 

I'm building a nextJS app around the CMS umbraco. Currently our setup is as follows:

_app.tsx _- fetches navigation and global metadata through getInitialProps_
[...index.tsx] _- catches the url slug and fetches layout data for the current page through getServersideProps_

Using <Link> i expected navigating between pages would only invoke the new pages getServersideProps, but it seems that if the page navigated to has a getServersideProps function _app.tsx'es getInitialProps()s function is invoked before hand as well. 

Reading through earlier discussions a lot of people suggest going for a static approach, but it's unfortunately not a viable option for us at the moment. 

Here's a mini example https://github.com/icew1nd/getinitialprops-test

If this is intended or neccessary, what do guys do to speed up your headless CMS implementations?

**Comment by icew1nd** [comment, marked as answer]
Alright - So i figured this out eventually;

So when using <Link> combined with a catch-all route like [...index.tsx] to get the desired effect you need to specificly point to the file you want to hit then use the `as` property.

So instead of 
`<Link href="/path/to/somehere">`

It should be:

`<Link href="/[...index]" as="/path/to/somehere">`

Hope this helps someone else in the future :- )

**Reply by thebuilder** [reply]
You should not have to use the `as` prop since 9.5.3. See #16634

I'd recommend you go with the static approach, and fetch layout (navigation, meta data) in the `getStaticProps` for the pages. `_app.js` can read it from the PageProps, and render the global layout accordingly. If you are going with serverside rendering, look into adding `stale-while-revalidate`, so you're caching the HTML response.
